InformationScripting/src/dataformat/Tuple.cpp

```diff
@@ -28,16 +28,17 @@
 
 namespace InformationScripting {
 
-Tuple::Tuple(std::initializer_list<NamedProperty> initialValues)
-	: values_{initialValues}
+Tuple::Tuple(std::initializer_list<NamedProperty> initialValues, QString tag)
+	: values_{initialValues}, tag_{tag}
 {}
 
-Tuple::Tuple(QList<NamedProperty> initialValues)
-	: values_{initialValues}
+Tuple::Tuple(QList<NamedProperty> initialValues, QString tag)
+	: values_{initialValues}, tag_{tag}
 {}
 
 QString Tuple::tag() const
 {
+	if (!tag_.isNull()) return tag_;
```

Comment:
Well, my only concern is that having this flexibility slightly complicates our basic data model. But I am not sure if that's indeed a problem or not.

Perhaps a slightly nicer solution is to say:
- A tuple always has a tag
- You can either provided manually
- Or else it gets automatically assgined at construction from the tag of the first tuple element. This is very similar to the current solution with two differences:
 - In the default case, we insist that the tuple is initialized with at least one value, so that the empty case at the bottom of this function falls out.
 - We change this method to always return `tag_` and inline it, and move the entire initialization to the initializer list in the constructors. This is a bit more explicit.

What do you think?